starter-core/src/main/java/io/micronaut/starter/feature/database/DataJpaReactive.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.database;
+
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.build.dependencies.Priority;
+import io.micronaut.starter.feature.FeatureContext;
+import io.micronaut.starter.feature.database.jdbc.JdbcFeature;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class DataJpaReactive implements JpaFeature, DataFeature {
```

Comment:
It looks confusing but Hibernate Reactive doesn't use JDBC! It only uses the same URL format. We would need to have something like `VertxDatabaseDriverFeature`.

Reply:
So we'd need 5 new features to support this? 🤔 

@sdelamo can you think of any other way of doing this?

Reply:
Or maybe we do it as for r2dbc and have just one driver and add it here:
```
public void apply(GeneratorContext generatorContext) {
        if (generatorContext.isFeaturePresent(R2dbc.class)) {
            generatorContext.addDependency(Dependency.builder()
                    .groupId("org.postgresql")
                    .artifactId("r2dbc-postgresql")
                    .runtime());
            if (!generatorContext.isFeaturePresent(MigrationFeature.class)) {
                return;
            }
        }

        generatorContext.addDependency(Dependency.builder()
                .groupId("org.postgresql")
                .artifactId("postgresql")
                .runtime());
    }
```
plus add `VertxDatabaseClientFeature`

Reply:
> Yes, and I think we need to rename `vertx-mysql-client`(and `vertx-pg-client`) to `micronaut-vertx-mysql-client`

That's the second option isn't it?  I thought we were going with the first option?

https://github.com/micronaut-projects/micronaut-starter/issues/1306#issuecomment-1172185927

Reply:
That's not correct.

We don't need:
```
    runtimeOnly("mysql:mysql-connector-java")
    implementation("io.micronaut.sql:micronaut-jdbc-hikari")
```

Instead of:
```
datasources:
  default:
    url: jdbc:mysql://localhost:3306/db
    driverClassName: com.mysql.cj.jdbc.Driver
    username: root
    password: ''
    schema-generate: CREATE_DROP
    dialect: MYSQL
```

Username, password + connection should be at:
```
jpa:
  default:
    properties:
      hibernate
```

Reply:
Ok, I have just [pushed a change](https://github.com/micronaut-projects/micronaut-starter/pull/1307#:~:text=f8e211e-,Skip%20regular%20datasource%20setup%20for%20hibernate%2Dreactive,-Verified) to the MySQL, MariaDB, etc classes which then gives us a config of:

```
micronaut:
  application:
    name: foo
datasources:
  default:
    schema-generate: CREATE_DROP
    dialect: MYSQL
jpa:
  default:
    properties:
      hibernate:
        hbm2ddl:
          auto: update
        connection:
          url: jdbc:mysql://localhost:3306/db
          username: root
          password: ''
    reactive: true
netty:
  default:
    allocator:
      max-order: 3
```

And a dependencies of:

```
    annotationProcessor("io.micronaut.data:micronaut-data-processor")
    annotationProcessor("io.micronaut:micronaut-http-validation")
    implementation("io.micronaut:micronaut-http-client")
    implementation("io.micronaut:micronaut-jackson-databind")
    implementation("io.micronaut.data:micronaut-data-hibernate-jpa")
    implementation("io.micronaut.sql:micronaut-hibernate-reactive")
    implementation("io.vertx:vertx-mysql-client")
    runtimeOnly("ch.qos.logback:logback-classic")
    testImplementation("org.testcontainers:junit-jupiter")
    testImplementation("org.testcontainers:testcontainers")
    implementation("io.micronaut:micronaut-validation")
```

🤞

Reply:
```
datasources:
  default:
    schema-generate: CREATE_DROP
    dialect: MYSQL
```
Is not needed.

Instead of:
```
  implementation("io.micronaut.data:micronaut-data-hibernate-jpa")
  implementation("io.micronaut.sql:micronaut-hibernate-reactive")
 ```
it should be:
```
  implementation("io.micronaut.data:micronaut-data-hibernate-reactive")
 ```
Or just SQL module if you don't select the data?

And testcontainers mysql module is missing.

Reply:
Ok, fixed I think 😅

Reply:
Gah!  Apart from testcontainers 😿